Networking/Networking/Remote/InAppPurchasesRemote.swift

```diff
@@ -4,6 +4,7 @@ import Foundation
 /// In-app Purchases Endpoints
 ///
 public class InAppPurchasesRemote: Remote {
+    public typealias InAppPurchasesTransactionResult = Swift.Result<InAppPurchasesTransactionResponse, Error>
```

Comment:
I thought about returning only the `siteID`, but since we may have 2 different objects returning a successful response (siteID or message/code) or an Error (decoding/networking/something else) it seemed simpler to return a `Result<Response, Error>` type and hide the details of what that result would be, leaving the business logic to handle it. 

Happy to iterate further if you see a better way to handle this!

Reply:
I see, could we then return `InAppPurchasesTransactionResponse` asynchronously from that function? If the function throws an error as well we don't need to return a result as far as I can tell.

Reply:
>  could we then return InAppPurchasesTransactionResponse asynchronously from that function?

Unless I miss-understood, we already do so via the [async wrapper here](https://github.com/woocommerce/woocommerce-ios/blob/ab3bf022ed87b5ae4a8f6c1cd1a84ccecdb83bda/Networking/Networking/Remote/InAppPurchasesRemote.swift#L100C3-L106), since we can't return the original method directly as async as this has to conform to the `Mapper` protocol.

```swift
    func retrieveHandledTransactionResult(for transactionID: UInt64) async throws -> InAppPurchasesTransactionResponse {
        try await withCheckedThrowingContinuation { continuation in
            retrieveHandledTransactionResult(for: transactionID) { result in
                continuation.resume(with: result)
            }
        }
    }
```

Let me know if you're referring to something else! 🙇

Reply:
Ah I see, then it's correct, thanks a lot for the answer 👍